test/metabase/api/public_test.clj

```diff
@@ -1276,3 +1278,46 @@
                                    (:public_uuid dash)
                                    dashcard-id
                                    (json/encode {:id 1})))))))))))
+
+;;; --------------------------------- POST /api/public/action/:uuid/execute ----------------------------------
+
+(deftest execute-public-action-test
+  (mt/with-actions-test-data-and-actions-enabled
+    (mt/with-temporary-setting-values [enable-public-sharing true]
+      (let [{:keys [public_uuid] :as action-opts} (shared-obj)]
+        (mt/with-actions [{} action-opts]
+          (testing "Check that we get a 400 if the action doesn't exist"
+            (is (= "An error occurred."
+                   (client/client
+                    :post 400
+                    (format "public/action/%s/execute" (str (UUID/randomUUID)))
+                    {:parameters {:id 1 :name "European"}}))))
+          (testing "Check that we get a 400 if sharing is disabled."
+            (mt/with-temporary-setting-values [enable-public-sharing false]
+              (is (= "An error occurred."
+                     (client/client
+                      :post 400
+                      (format "public/action/%s/execute" public_uuid)
+                      {:parameters {:id 1 :name "European"}})))))
+          ;; TODO: this needs to pass when this PR is merged https://github.com/metabase/metabase/pull/27716
+          #_(testing "Check that we get a 400 if actions are disabled for the database."
```

Comment:
Note I plan to uncomment this in another PR. At the moment this is a bug in general: actions can get executed even if the database has actions disabled. 

When https://github.com/metabase/metabase/pull/27716 is completed, I will merge it here, uncomment this test, and fix it.